bench/workgen/workgen.cpp

```diff
@@ -555,17 +555,25 @@ WorkloadRunner::select_table_for_drop(std::vector<std::string> &pending_delete)
     ContextInternal *icontext = _workload->_context->_internal;
     auto table_itr = icontext->_dyn_tint.begin();
     std::advance(table_itr, workgen_random(_rand_state) % icontext->_dyn_tint.size());
+
+    bool has_mirror = icontext->_dyn_table_runtime[table_itr->second].has_mirror();
+    auto mirror_itr = has_mirror ?
+      icontext->_dyn_tint.find(icontext->_dyn_table_runtime[table_itr->second]._mirror) :
+      icontext->_dyn_tint.end();
+
     if (icontext->_dyn_table_runtime[table_itr->second]._pending_delete) {
+        // The mirror should have been marked as pending deletion too.
+        if (has_mirror)
+            ASSERT(mirror_itr != icontext->_dyn_tint.end() &&
+              icontext->_dyn_table_runtime[mirror_itr->second]._pending_delete);
```

Comment:
Maybe we should move to this assert, into when we are doing an table operation or before we commit a transaction.

Reply:
After checking the code, moving the check would be wrong. A table can be marked as pending deletion while threads are working on it even though they are not done with their transaction.
When marked for deletion, no new threads can use the table.